### DISCUSSION TITLE
Target typing should work while performing record nondestructive mutation.

### DISCUSSION THREAD

**Discussion by HobbsCode** [dotnet/csharplang#4822] [General]
**Version Used**: 
.NET 5

**Steps to Reproduce**:
[SharpLab](https://sharplab.io/#v2:D4AQTAjAsAULIGYAEAnApgYwPYoCZIAU0UBnLAOyVgG8qYkGlEmIAGJAOQEMBbNJagHM0AFwDcSAJblJ4gL6M6c2PGTgWAdhqNYi8GDqLquxYyKkKSLgBtJGfgF4k5NAHdCxMuQAUASiSusgAWApy8jkgARDZ2aJFIchIA9ElIAPIA0iamHhaUAEZY+UhOLq5+AcGh3HwlUYX58YlIKUgAoigoOEgAwgDKABwaAKwALNmmE3oQAJzeMfa+YlOMILPeDUsTCrByQA)

```
public record Person 
{ 
    public string Name {get; init;}    
}

public class Program
{   
    public static void Main() 
    {
        Person alice = new Person() with { Name = "alice" }; // OK
        Person bob = new() with { Name = "bob" }; // Error CS8754
        
        Console.WriteLine(alice);
        Console.WriteLine(bob);
    } 
}
```

**Expected Behavior**:
The program should compile without error. 

**Actual Behavior**:
The compiler rejects the initialization of variable 'bob' with CS8754.

**Comment by CyrusNajmabadi** [comment]
I'm not getting the value here.  if Person is a record, this would already work and woudl be shorter and clearer:

```c#
Person bob = new() { Name = "bob" };
```

**Comment by Joe4evr** [comment]
If I had a stake in this, I'd consider `new Person() with { .... }` to be a full-on compiler error as well:
1. It's double-allocating an object for completely no reason
2. It can actually break at runtime if/once the language gets that "final initializer" feature and the `Person` class adds one of those to throw if `Name` isn't set.

**Reply by HobbsCode** [reply]
One is legal while the other is not. Seems like they should be treated the same. I doubt anyone has taken any meaningful dependency on this behavior.

**Reply by Joe4evr** [reply]
True, though it's up to the LDT to determine if they want to make that happen.
